qiskit_aer/quantum_info/states/aer_statevector.py

```diff
@@ -184,27 +184,56 @@ def _aer_evolve_circuit(aer_state, circuit, qubits):
     @staticmethod
     def _aer_evolve_instruction(aer_state, inst, qubits):
         """Apply instruction into aer_state"""
+
         params = inst.params
+
+        def _assert_qubits(expected):
+            if len(qubits) != expected:
+                raise AerError(
+                    f"Cannot apply a {expected}-qubit instruction {inst} to qubits {qubits}"
+                )
+
+        def _assert_params(expected):
+            if len(params) < expected:
+                raise AerError(f"Need {expected} parameters, but actually {len(params)}")
+
         if inst.name in ['u', 'u3']:
-            aer_state.apply_mcu(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1],
-                                params[0], params[1], params[2])
-        elif inst.name in ['x', 'cx', 'ccx']:
-            aer_state.apply_mcx(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1])
-        elif inst.name in ['y', 'cy']:
-            aer_state.apply_mcy(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1])
-        elif inst.name in ['z', 'cz']:
-            aer_state.apply_mcz(qubits[0:len(qubits) - 1], qubits[len(qubits) - 1])
+            _assert_qubits(1)
+            _assert_params(3)
+            aer_state.apply_u(qubits[0], params[0], params[1], params[2])
+        elif inst.name == 'h':
+            _assert_qubits(1)
+            aer_state.apply_h(qubits[0])
+        elif inst.name == 'x':
+            _assert_qubits(1)
+            aer_state.apply_x(qubits[0])
+        elif inst.name == 'cx':
+            _assert_qubits(2)
+            aer_state.apply_cx(qubits[0], qubits[1])
+        elif inst.name == 'y':
+            _assert_qubits(1)
+            aer_state.apply_y(qubits)
+        elif inst.name == 'cy':
+            _assert_qubits(2)
+            aer_state.apply_cy(qubits[0], qubits[1])
+        elif inst.name == 'z':
+            _assert_qubits(1)
+            aer_state.apply_z(qubits)
+        elif inst.name == 'cz':
+            _assert_qubits(2)
+            aer_state.apply_cz(qubits[0], qubits[1])
         elif inst.name == 'unitary':
             aer_state.apply_unitary(qubits, inst.params[0])
         elif inst.name == 'diagonal':
             aer_state.apply_diagonal(qubits, inst.params[0])
         elif inst.name == 'reset':
             aer_state.apply_reset(qubits)
+        elif inst.name in ['barrier', 'id']:
+            pass
         else:
             definition = inst.definition
-            if definition is inst:
-                raise AerError('cannot decompose ' + inst.name)
-            if not definition:
+            if definition is inst or definition is None:
+                print(inst.definition)
```

Comment:
`print` probably isn't the right choice here - is it a left-over debugging statement?

Reply:
sorry. it is just for debugging.